pkgs/build-support/php/hooks/composer-install-hook.sh

```diff
@@ -0,0 +1,77 @@
+declare -a composerVendorCache
+declare -a installComposerDevDependencies
+declare -a runComposerScripts
+declare -a runComposerPlugins
+declare -a dontComposerInstallConfigureHook
+declare -a dontComposerInstallBuildHook
+declare -a dontComposerInstallInstallHook
+declare -a version
+
+if [[ -z "${dontComposerInstallConfigureHook-}" ]]; then
+  preConfigureHooks+=(composerInstallConfigureHook)
+fi
+
+if [[ -z "${dontComposerInstallBuildHook-}" ]]; then
+  preBuildHooks+=(composerInstallBuildHook)
+fi
+
+if [[ -z "${dontComposerInstallInstallHook-}" ]]; then
+  preInstallHooks+=(composerInstallInstallHook)
+fi
+
+composerInstallConfigureHook() {
+    echo "Executing composerInstallConfigureHook"
+
+    if [[ ! -e "${composerVendorCache}" ]]; then
+        echo "No composer cache found."
+        exit 1
+    fi
+
+    cp -fr ${composerVendorCache}/. .
+    chmod +w .composer -R
+
+    echo "Finished composerInstallConfigureHook"
+}
+
+composerInstallBuildHook() {
+    echo "Executing composerInstallBuildHook"
+
+    argstr=("--no-interaction" "--optimize-autoloader")
+
+    if [[ ! -n ${installComposerDevDependencies-} ]]; then
+        argstr+=("--no-dev")
+    fi
+
+    if [[ ! -n ${runComposerScripts-} ]]; then
+        argstr+=("--no-scripts")
+    fi
+
+    if [[ ! -n ${runComposerPlugins-} ]]; then
+        argstr+=("--no-plugins")
+    fi
+
+    argstr+=("--apcu-autoloader-prefix=nixPredictableApcuAutoloaderPrefix")
```

Comment:
The `--apcu-autoloader-prefix` implicitly enables `--apcu-autoloader` so we might want to only enable it when `--apcu-autoloader` is enabled via the configuration. Alternatively we could make sure the APCU autoloader is never enabled as it does not make a lot of sense in our situation (you are not going to write new files at runtime in the Nix store)


https://getcomposer.org/doc/03-cli.md#install-i

Reply:
Since I removed it and added the possibility to override the composer.json file, I guess I can consider this as done.